File: syno_backend/src/main/java/com/syno_back/backend/dto/UpdateRequestDto.java
package com.syno_back.backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import java.util.List;

/**
 * DTO for accepting update request from client
 */
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Builder
@ToString
public class UpdateRequestDto {
    /**
     * List of existing pins
     * Future pagination-friendly
     */
    @JsonProperty("existing_dict_pins")
    private List<String> existingDictPins;

    /**
     * List of updated dicts
     */
    @JsonProperty("client_dicts")
    private List<UpdateUserDictionaryDto> clientDicts;
}

File: syno-mobile/syno-mobile/Core/Dtos/UpdateRequestDto.swift
import Foundation

/// DTO for sending request for creating copy on server
class UpdateRequestDto: Codable {
    /// List of dict pins on client
    var existingDictPins: [String]
    
    /**
     List of client dictionaries
     
     May not contain all dictionaries for pagination
     */
    var clientDicts: [UpdateUserDictionaryDto]
    
    /**
     Creates new `UpdateRequestDto`
     */
    init(existingDictPins: [String], clientDicts: [UpdateUserDictionaryDto]) {
        self.existingDictPins = existingDictPins
        self.clientDicts = clientDicts
    }
}

